pkg/identityserver/api_key_utils.go

```diff
@@ -43,11 +44,22 @@ func GenerateAPIKey(ctx context.Context, name string, rights ...ttnpb.Right) (ke
 	if err != nil {
 		return nil, "", err
 	}
-	key = &ttnpb.APIKey{
-		ID:     generatedID,
-		Key:    hashedKey,
-		Name:   name,
-		Rights: rights,
+	// Check that expiry date passed is not a nil value
+	if expiresAt != nil {
+		key = &ttnpb.APIKey{
+			ID:        generatedID,
+			Key:       hashedKey,
+			Name:      name,
+			Rights:    rights,
+			ExpiresAt: expiresAt,
+		}
+	} else {
+		key = &ttnpb.APIKey{
+			ID:     generatedID,
+			Key:    hashedKey,
+			Name:   name,
+			Rights: rights,
+		}
```

Comment:
Unless you also meant to check for `expiresAt.IsZero()`, this isn't needed.

Reply:
ah yea this is a leftover from when there was a zero value